# Лабораторная работа: Написание вашего первого кода Python

## Цели

* Написать базовый код в Python
* Работать с различными типами данных в Python
* Конвертировать данные из одного типа в другой
* Использовать выражения и переменные для выполнения операций

## При использовании Jupiter notebook не обязательно писать print для вывода данных
***Если вы работаете в через стандартный файл .py, то это делать необходимо***

<h2> Содержание </h2>
<div class = "rewert-lert-block alert-info" style = "margin-top: 20px">
    <ul>
        <li>
            <a href="#hello"> Скажите «привет» миру в Python </a>
            <ul>
                <li> <a href="comments"> написание комментариев в Python </a> </li>
                <li> <a href="errors"> ошибки в Python </a> </li>
                <li> <a href="python_error"> Python знает о вашей ошибке, прежде чем запустит ваш код? </a> </li>
                <li> <a href="exercise"> упражнение: ваша первая программа </a> </li>
            </ul>
        </li>
        <li>
            <a href="#types_objects"> типы объектов в Python </a>
            <ul>
                <li> <a href="int"> целые числа </a> </li>
                <li> <a href="float"> плавает </a> </li>
                <li> <a href="convert"> преобразование из одного типа объекта в другой тип объекта </a> </li>
                <li> <a href="bool"> логический тип данных </a> </li>
                <li> <a href="exer_type"> упражнение: типы </a> </li>
            </ul>
        </li>
        <li>
            <a href="#expressions"> выражения и переменные </a>
            <ul>
                <li> <a href="exp"> выражения </a> </li>
                <li> <a href="exer_exp"> упражнение: выражения </a> </li>
                <li> <a href="var"> переменные </a> </li>
                <li> <a href="exer_exp_var"> упражнение: выражение и переменные в Python </a> </li>
            </ul>
        </li>
    </ul>
</div>

<hr>


<h2 id = "hello"> Скажи "Привет" миру на Python </h2>


При изучении нового языка программирования обычно начинается с примера «Hello World».Как бы это ни было, эта одна строка кода обеспечит, чтобы мы знали, как печатать строку в выводе и как выполнять код в ячейках в ноутбуке.


<hr/>
<div class = "оповещение о предупреждении AlertSuccess" style = "margin-out: 20px">
Чтобы выполнить код Python в ячейке кода ниже, нажмите на ячейку, чтобы выбрать ее и нажмите <KBD> SHIFT </KBD> + <KBD> ENTER </KBD>.
</div>
<hr/>


In [ ]:
# Выведите свою первую фразу на Python

print('Hello, Python!')


<hr/>
<div class = "оповещение о предупреждении AlertSuccess" style = "margin-out: 20px">
<code> print () </code> - функция.Вы передали строку <code> 'Hello, Python!' </Code> в качестве аргумента для обучения Python о том, что нужно печатать (выводить пользователю).
</div>
<hr/>


<h3 id = "Комментарии"> Написание комментариев в Python </h3>

<p>
    В дополнение к написанию кода, обратите внимание, что всегда полезно добавить комментарии к вашему коду.Это поможет другим понять, чего вы пытались достичь (причина, по которой вы написали данный фрагмент кода).Это не только помогает <strong> другим людям </strong> понять ваш код, это также может служить напоминанием <strong> для вас </strong>, когда вы вернетесь к нему через несколько недель или месяцев. </P>
<p>
    Чтобы написать комментарии в Python, используйте символ номера <code>#</code> перед написанием вашего комментария.Когда вы запускаете свой код, Python игнорирует все после <code>#</code> в данной строке.
</p>
    Так же возможны многострочные комментарии, которые заключаются в тройные кавычки
    Закомментированные строки никак не влияют на код

In [ ]:
# Написание комментариев

print('Hello, Python!') # Однострочный комментарий
# print("hi")

<p>
    После выполнения ячейки выше, вы должны заметить, что <code> Эта строка печатает string </code> не появляется в выводе, потому что это был комментарий (и, таким образом, игнорируется Python).
</p>
<p>
    Вторая строка также не была выполнена, потому что <code> print ('hi') </code> также предшествовал знак номера (<code>#</code>) также!Поскольку это не объяснительный комментарий от программиста, а фактическая строка кода, мы могли бы сказать, что программист <em> прокомментировал </em> вторую строку кода.
</p>


<h3 id="errors">Ошибки в Python</h3>


<p> Все делают ошибки.Для многих типов ошибок Python скажет вам, что вы допустили ошибку, предоставив вам сообщение об ошибке.Важно тщательно прочитать сообщения об ошибках, чтобы по -настоящему понять, где вы допустили ошибку и как вы можете исправить их. </P>
<p> Например, если вы пишете  не <code> print </code> а <code> frint </code>, Python отобразит сообщение об ошибке.Попробуйте: </p>


In [9]:
# Запустите ячейку чтобы увидеть сообщение об ошибке

frint("Hello, Python!")

NameError: name 'frint' is not defined

<p> Сообщение об ошибке говорит вам:
<ol>
    <li> Где произошла ошибка (более полезная в больших ячеек ноутбука или сценариев) и </li>
    <li> Какая это была ошибка (nameerror) </li>
</ol>
<p> Здесь Python попытался запустить функцию <code> frint </code>, но не смог определить, что <code> это за команда. </p>


<p>
Вы заметите, что если мы сделаем другой тип ошибки, забыв закрыть строку, мы получим другую ошибку (то есть, <code> syntaxError </code>). Запустите ячейку ниже:
</p>


In [10]:
# Для того чтобы увидеть built-in сообщение об ошибке

print("Hello, Python!)

SyntaxError: unterminated string literal (detected at line 3) (Temp/ipykernel_3236/3628399183.py, line 3)

<h3 id = "python_error"> знает ли Python о вашей ошибке, прежде чем запустить ваш код? </h3>


Python - это то, что называется <em> интерпретируемым языком </em>.Скомпилированные языки изучают всю вашу программу во время компиляции и могут предупредить вас о целом классе ошибок до выполнения.

Напротив, Python интерпретирует вашу линию скрипта по линии, когда он выполняет ее.

Python перестанет выполнять всю программу, когда она столкнется с ошибкой (если ошибка не ожидается и обрабатывается программистом, 

более продвинутой темой, которую мы рассмотрим позже в этом курсе).


Попробуйте запустить код в ячейке ниже и посмотрите, что произойдет:


In [11]:
# Интерпретатор Python читает код сверху вних и слева направо
# Поэтому первая строка кода выведется без ошибки. а вторая заставит Python остановить программу

print("This will be printed")
frint("This will cause an error")
print("This will NOT be printed")

This will be printed


NameError: name 'frint' is not defined

<h3 id="exercise">Упражнение: ваша первая программа</h3>


<p> поколения программистов начали свою карьеру в кодировании, просто печатая «Привет, мир!».Вы будете идти по их стопам. </P>
<p> В кодовой ячейке ниже используйте функцию <code> print () </code>, чтобы распечатать фразу: <code> Привет, мир! </code> </p>


In [ ]:
# Напишите свой код ниже. Не забудьте нажать Shift+Enter, чтобы выполнить ячейку


<details><summary>Нажмите здесь для решения</summary>

```python
print("Hello, world!")

```

</details>


<P> Теперь давайте улучшим ваш код с помощью комментария. В кодовой ячейке ниже распечатайте фразу: <code> Привет, мир!


In [ ]:
# Напишите свой код и закомментируйте его


<details><summary>Нажмите здесь для решения</summary>

```python
print("Hello, world!") # Print the traditional hello world

```

</details>


<hr>


<h2 id="types_objects" align="center">Типы обьектов в Python</h2>


<p> Python-это объектно-ориентированный язык.В Python есть много разных типов объектов.

Давайте начнем с наиболее распространенных типов объектов: <i> Strings (строк) </i>, <i> integers (целых чисел) </i> и <i> floats (чисел с плавающей запятой) </i>.В любое время, когда вы пишете слова (текст) в Python, вы используете строчные символы </i> (строки).

А вот наиболее распространенными числами, являются <i> целые числа </i> (например, -1, 0, 100) и <i> floats </i>, которые представляют реальные числа (например, 3.14, -42.0).</p>


<a align="center">
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0101EN-SkillsNetwork/labs/Module%201/images/TypesObjects.png" width="600">
</a>


<p>Следующие кодовые ячейки содержат несколько примеров.</p>


In [22]:
# Integer
11


11

In [24]:
# Для файла .py
print(11)

11


In [13]:
# Float. Для файла .py  print(2.14)
2.14

2.14

In [14]:
# String. Для файла .py  print("Hello, Python 101!")

"Hello, Python 101!"

'Hello, Python 101!'

<p> Вы можете попросить Python, чтобы он сообщил вам тип возвращаемого выражения или переменной, используя встроенную функцию <code> type () </code>.

Вы заметите, что Python называет целые числа как <code> int </code>, числа с плавающей запятой как <code> float </code> и строки символов как <code> str </code>. </P>


In [15]:
# Тип 12
type(12)


int

In [25]:
# для стандартного типа файлов .py
print(type(12))# функция type() и ее аргумент - 12, помещаются в функцию print()
# вывод говорит что эт ообьект класса int - целых чисел

<class 'int'>


In [16]:
# Tип 2.14
type(2.14)


float

In [17]:
# Tип выражения "Hello, Python 101!"
type("Hello, Python 101!")


str

<p> В ячейке кода ниже используйте функцию <code> type () </code>, чтобы проверить тип объекта <code> 12.0 </code>.


In [ ]:
# Напишите свой код ниже. Не забудьте нажать Shift+Enter, чтобы выполнить ячейку



<details><summary>Нажмите здесь для решения</summary>

```python
type(12.0)

```

</details>


<h3 id="int">Integers. Целые числа</h3>


<p>Вот несколько примеров целых чисел. Целые числа могут быть отрицательными или положительными числами:</p>


<a align="center">
    <img src="https://cf-courses-data.s3.us.cloud-object-storage.appdomain.cloud/IBMDeveloperSkillsNetwork-PY0101EN-SkillsNetwork/labs/Module%201/images/TypesInt.png" width="600">
</a>


<p>Мы можем убедиться, что это так, используя, как вы уже догадались, <code> type () </code>:

При использовании стандартных файлов .py необходимо включать эти функции в функцию print() для вывода

<code> print(your_function(arguments)) </code>

In [18]:
# вывести на экран тип данных -1
type(-1)

int

In [19]:
# Вывести на экран тип данных 4
type(4)

int

In [ ]:
# Print the type of 0
type(0)

<h3 id="float">Floats. Числа с плавающей запятой</h3> 


<P> Поплавки представляют собой реальные числа (Real numbers);

Они представляют собой суперсет целочисленных чисел, но также включают «цифры с десятиками».

Есть некоторые ограничения, когда речь идет о машинах, представляющих реальные числа, но в большинстве случаев плавучие номера являются хорошим представлением.

Вы можете узнать больше о специфике поплавков для вашей среды выполнения, проверяя значение <code> sys.float_info </code>.

До точки идет наименование используемой библиотеки (sys) - ссылка на библиотеку https://docs.python.org/3/library/sys.html

Это также скажет вам, какое самое большое и наименьшее число может быть представлено с ними. </P>

<p> Еще раз, может проверить некоторые примеры с помощью <code> type () </code> function:


In [ ]:
# Print the type of 1.0

type(1.0) # 1 это int, но 1.0 это float

In [27]:
# Для того чтобы посмотреть информацию о любом классе или типе данных сможно воспользоваться библиотекой sys
import sys # ждя того чтобы библиотека работала ее нужно импортировать. но об этом позднее
type(1)
sys.float_info
# или print(sys.float_info)

sys.float_info(max=1.7976931348623157e+308, max_exp=1024, max_10_exp=308, min=2.2250738585072014e-308, min_exp=-1021, min_10_exp=-307, dig=15, mant_dig=53, epsilon=2.220446049250313e-16, radix=2, rounds=1)

<h3 id="convert">Преобразование из одного типа объекта в другой тип объекта</h3>


<p>Вы можете изменить тип объекта в Python;

Это называется ***TypeCasting***. Например, вы можете преобразовать <i> integer </i> в <i> float </i> (например, 2 до 2.0). </P>
<p> Попробуем:</p>


In [31]:
# Запомните , что первоначальный тип аргумента (2) - это integer
type(2)


int

<h4>Преобразование целых чисел в числа с плавающей запятой (поплавки) </h4>
<p> Давайте возьмем целое число 2, и сменим его тип (для повышения точности вычислений): </p>


In [32]:
# Конвертация 2 в тип данных float
float(2)


2.0

**Следующую операцию можно выполнять поэтапно. То есть, в ячейке написано слeдующее**:

* Преобразовать 2 в число с плавающей запятой -> функция вернет 2.0
* Передать 2.0 в качестве аргумента функции type() -> функция вернет тип данных
* Для вывода данных в консоль при использовании стандартного файла .py необходимо использовать <code> print(*остальные функции, результат которых необходимо вывести*)<code> 


In [ ]:
# проверим тип данных возвращаемый конвертицией (передадим функции type() в качестве аргумента, функцию преобразования и ее агрумент)
type(float(2))

<p> Когда мы конвертируем целое число в поплавок, мы на самом деле не меняем значение (то есть значимость) числа.
Однако, если мы переводим поплавок (float) в целое число (int), мы могли бы потерять некоторую информацию.

Например, если мы конвертируем Float 1.1 в целое число, мы получим 1 и потеряем десятичную информацию (т.е. 0,1): </p>


In [33]:
# Проверьте это утверждение (преобразование дробного числа в целое)

int(1.1)


1

### ***Преобразование из строк в целые числа или поплавки***


<p>Иногда мы можем иметь строку, которая содержит число в ней.

Если это так, мы можем конвертировать эту строку, которая представляет число в целое число, используя <code>int()</code>:</p>


In [34]:
# Конвертируйте  строку в целое число
# Если вы ставите число в кавычки, то это строка

int('1')


1

<center><h4>ОШИБКА КОНВЕРТИЦИИ</h4></center>

<h4>
<p>Но, если вы попытаетесь сделать это со строкой, которая не идеально подходит для числа, вы получите ,<code>ошибку</code>

Попробуйте следующее:</p><h4>


In [35]:
# Получите ошибку конвертиции
int('1 or 2 people')

# иногда такие моменты становятся краеугольными, особенно при получении пользовательского ввода, который необходимо интерпретировать как число
#Например для последующих математический вычислений



ValueError: invalid literal for int() with base 10: '1 or 2 people'

<p> Вы также можете преобразовать строки, содержащие номера с плавающими запятыми и в объекты <i> float </i>: </p>


In [ ]:
# конвертируйте строку "1.2" в тиип данных float

float('1.2')

<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
[Tip:] Обратите внимание, что строки могут быть представлены с помощью отдельных кавычек (<code> '1.2' </code>) или двойными кавычками (<code> "1.2" </code>), но вы не можете смешать оба (например,, <code> "1.2 '</code>).
</div>
<hr/>


<h4>Преобразование чисел в строки</h4>


<p> Если мы можем преобразовать строки в числа, естественно предположить, что мы можем преобразовать числа в строки, верно? </p>


In [36]:
# Конвертируем и сразу проверим тип возвращаемых данных (как в математике сначала выполняются действия в скобках)
# 
type(str(1))


str

<p> И нет никаких причин, по которым мы не должны быть в состоянии делать поплавки в строки: </p>


In [37]:
# Кронвертируем float в string

str(1.2)

'1.2'

<h3 id="bool">Логический тип данных</h3>


<p><i>Boolean</i> является еще одним важным типом в Python. Объект типа <i> boolean (логический) </i> может принять одно из двух значений: <code>True</code> или <code>False</code>:</p>


In [ ]:
# Истинное значение

True

<p> Обратите внимание, что значение <code> True </code> написано с большой буквы "T".То же самое относится и к <code> False </code> (т.е. вы должны использовать верхний кейс "F"). </P>


In [38]:
# Ложное значение
False

False

<p> Когда вы попросите Python отобразить тип логического объекта, он покажет <cod> bool </code>, который означает <i> boolean </i>: </p>

In [ ]:
# Тип данных True
type(True)


In [ ]:
# тип данных False
type(False)


<p> Мы можем конвертировать логические объекты в другие типы данных. Если мы попробуем конвертировать логическое значение со значением <code> True </code> в целое число или поплавок, мы получим 1.

Если мы конвертируем логическое значение <code> false </code> на целое число или поплавок, мы получим 0 (ноль).

Точно так же, если мы конвертируем 1 (целое число) на логическое, получим <code> True </code>. И если мы конвертируем 0 на bool, мы получим <code> False </code>.

Давайте попробуем: </p>


In [40]:
# конвертируем True в int
int(True)


1

In [ ]:
# конвертируем 1 в bool

bool(1)

In [ ]:
# конвертируем 0 в bool

bool(0)

In [ ]:
# конвертируем True в float

float(True)

<h3 id="exer_type">Упражнение: Типы данных</h3>


<p>Каков тип данных выражения: <code>6 / 2</code>?</p>


In [ ]:
# Напишите свой код ниже. Не забудьте нажать Shift+Enter, чтобы выполнить ячейку


<details><summary>Нажмите здесь для решения</summary>

```python
type(6/2) # float
# or
print(type(6/2))
```

</details>


<p>Каков тип результата: <code> 6 // 2 </code>? (Обратите внимание на двойную черту <code>//</code>)</p>


In [ ]:
# Напишите свой код ниже. Не забудьте нажать Shift+Enter, чтобы выполнить ячейку


<details><summary>Нажмите здесь для решения</summary>

```python
type(6//2) # int, двойной слэш обозначает целочесленное деление
# или для стандартного файла .py
print(type(6//2))

```

</details>


<hr>


<center><h2 id="expressions">Выражения и переменные</h2></center>


<h3 id="exp">Expressions (Выражения) </h3>


<p>Выражения в Python могут включать операции между совместимыми типами (например, целыми и поплавками).

Например, основные арифметические операции, такие как сложение, вычитание нескольких чисел:</p>


In [41]:
# Простоая операция по сложению чисел 

43 + 60 + 16 + 41 


160

В Python существует несколько вариантов работы с выражениями
* через интерпретатор из командной строки: ввести в командную строку ключевое слово python, далее ввести выражение, нажать Enter
* через сохраниение результата в переменной, если сохраненный результат нужен для дальнейшей работы: 

<code>result = 43 + 60 + 16 + 41
print(result)</code>

* Или через вывод результата выражения без присвоения его переменной: print(43 + 60 + 16 + 41)

<p>Мы можем выполнить операции вычитания, используя оператор минус.В этом случае результатом является отрицательное число:</p>


In [42]:
#  операция вычитания

50 - 60

-10

<p>Мы можем сделать умножение, используя звездочку:</p>


In [ ]:
# Операция умножения

5 * 5

<p>Мы также можем выполнить деление с прямой слэш:


In [ ]:
# Деление

25 / 5

In [ ]:
# Результатом деления двух цедлых чисел будет целое число

25 / 6

<p>Для явного указания на то, что мы используем целочисленное деление мы можем использовать двойную черту , где результат округлен до ближайшего целого числа:


In [ ]:
# Целочисленное деление

25 // 5

In [ ]:
# целочисленное деление

25 // 6

<h3 id="exer_exp">Упражнения: Выражения</h3>


<p>Давайте напишем выражение, которое вычисляет, сколько часов есть за 160 минут:


In [ ]:
# Напишите свой код ниже. Не забудьте нажать Shift+Enter, чтобы выполнить ячейку (или нажмите стрелочку слева от ячейки)


<details><summary>Нажмите здесь для решения</summary>

```python
160/60 

# или

160//60

# для .py
print(160//60)
# или
hours = 160//60
print(hours)
```

</details>


<p>Python следует хорошо принятым математическим соглашениям при оценке математических выражений.В следующем примере Python добавляет 30 к результату умножения (то есть 120).


In [43]:
# Математическое выражение

30 + 2 * 60

150

<p>И так же, как математика, выражения, заключенные в скобки, имеют приоритет.Таким образом, следующие умножения 32 на 60.


In [ ]:
# Mathematical expression

(30 + 2) * 60

<h3 id="var">Variables (Переменные)</h3>


<p>Как и в случае с большинством языков программирования, мы можем хранить рузельтаты, возвращаемые функциями, математическими и иными выраженими в <i>переменных</i>, 
для того чтобы в дальнейшем использовать этот результат. 

Переменная, это именованная ячейка в памяти, в которую сохраняется та или иная информация к которой в последствии мы можем обрадаться по имени (имени переменной)

Рекомендуется заводить переменные только в случае. если потом требуется обращаться к этой информации. 
Имена перемнных должны соответствовать смысловой нагрузке получаемого результата

Например: Мы считаем ***количество фруктов*** и помещаем это в переменную, которую называем **fruints_quantity**</p>


In [44]:
# Хранение результатов в переменной

result = 43 + 60 + 16 + 41

<p>Чтобы увидеть значение <code> result </code> в ноутбуке, мы можем просто поместить его на последнюю строку ячейки
а для .py, как всегда, добавим функцию print():</p>


In [45]:
# Вывести результат вычисления

result

160

<p>Мы также можем выполнять дальнейшие операции c <code>result</code> и сохранить результат в новой переменной:</p>


In [46]:
# Используем другую переменную (division) для хранения результата деления переменной result на 60

division = result / 60
division
# если бы мы не использовали переменную resultБ в которой хранится результат выполнения сложения (43 + 60 + 16 + 41)
# наше выражение выглядело бы так: division = (43 + 60 + 16 + 41)/60

2.6666666666666665

<p>Если мы сохраним значение существующей переменной, новое значение перезаписывает предыдущее значение:</p>

Обратите внимание, что знак "=" в программировании означает присвоиение, запись в ячейку, ее перезапись, 

в то время, как равно(эквивалентность) - это знак "==" (двойное равно)


In [47]:
# Переприсвоение значения переменной result. теперь в этой ячейке записана новыя информация

result = result / 60
result

2.6666666666666665

<p>Это хорошая практика для использования значимых имен переменных, чтобы вы и другие могли прочитать код и понять его легче:</p>


In [48]:
# Предположим мы посмотрели 3 фильма, нам нужно узнать сколько времени в минутах мы на это потратили

total_min = 43 + 42 + 57 # общая продолжительность фильмов в минутах
total_min # наименование переменной самодокументируемое - то есть отвечает на вопрос, что мы в ней храним

142

In [49]:
# Самодокументируемые имена переменных

total_hours = total_min / 60 # общая продолжительность фильмов в минутах
total_hours

2.3666666666666667

<p> В ячейках выше мы добавили продолжительность трех фильмов в минутах и сохранили ее в <code> total_min </code>. Затем мы разделили его на 60, чтобы вычислить общую продолжительность в часах - <code> total_hours </code>.

Вы также можете сделать все это сразу в одном выражении, если вы используете скобку, как показано ниже. </P>


In [ ]:
# Усложненное выражение

total_hours = (43 + 42 + 57) / 60  # Вычисление длительности 3 фильмов в часах
total_hours

<p>Если вы предпочитаете общее количество часов в качестве целого числа, вы, конечно, можете заменить обычное деление на целочисленное (т.е.., <code>//</code>).</p>


<h3 id="exer_exp_var">Упражнения: выражение и переменные в Python</h3>


<p>Какой результат <code>x</code> где <code> x = 3 + 2 * 2</code></p>


In [ ]:
# Напишите свой код


<details><summary>Нажмите здесь для решения </summary>

```python
7

```

</details>


<p>Каково значение <code> y </code> где <code>y = (3 + 2) * 2</code>?</p>


In [ ]:
# Напишите свой код


<details><summary>Нажмите здесь для решения</summary>

```python
10

```

</details>


<p>Каково значение <code> z </code> где <code>z = x + y</code>?</p>


In [ ]:
# Напишите совй код


<details><summary>Нажмите здесь для решения</summary>

```python
17

```

</details>


## Обьявление переменных

Для создания переменной в Python нужно присвоить ей значение. При этом, если переменная еще не существует заранее, она будет создана автоматически.

Так же в переменной можно хранить пользовательский ввод

Повторение: Переменная в Python - это именованное место в памяти, которое хранит значение определенного типа данных. 

<code>В Python переменные не нуждаются в объявлении типа данных, так как тип данных определяется автоматически во время присваивания значения переменной</code>.

In [ ]:
# обьявление переменной для хранения будущего результата какой то операции
count = 0 # чаще всего таким образом обьявляются итераторы

price = input("Введите цену на товар:") # в переменную соххранится ответ пользователя на просьбу ввести цену
discount = 0.2 # фиксированная скидка (20%)
# инициализация переменной сразу с выражением
total_price = price - discount * price

In [ ]:
x = 5 # переменная хранит число
greeting = "Hello, world!" # переменная хранит строку

<h4> Объявление (инициализация) нескольниких переменных одной строкой</h4>

In [54]:
# Объявление нескольких переменных одного типа, ну и значения
x = y = z = 10
print(x)
print(y)
print(z)

10
10
10


In [57]:
# значения присваиваются в том же порядке
fruit1, fruit2, fruit3 = "Orange", 10, "Cherry"
print(fruit1)
print(fruit2)
print(fruit3)

# или
print(fruit1, fruit2, fruit3)

Orange
10
Cherry
Orange 10 Cherry


**Упражнение:**
Заведите 3 переменных и присвойте им разные значения но в одну строку. выведите на экран

In [ ]:
# Напишите свой код



## Зона видимости переменных

Зона видимости переменных в Python определяет область, в которой переменная может быть использована. В Python есть два типа переменных: локальные и глобальные.

То есть, если мы объявляем переменную внутри "закрытого куска кода", функция или цикл - она видна только в той области

при обьявлении переменной в общем коде - она видна повсеместно.

Локальные переменные - это переменные, которые определены внутри функции и могут быть использованы только внутри этой функции. Они не могут быть использованы за пределами функции. 

Пример:

In [ ]:
# закрытый участок кода
def my_function():
    x = 10
    print(x)
    
# если я захочу работать с переменной х вне функции - это будет иная переменная
my_function() # Выведет 10
print(x) # Ошибка, так как x не определена вне функции

<hr/>
<div class="alert alert-success alertsuccess" style="margin-top: 20px">
[Tip:] Обратите внимание, что обьявлять все переменные а начале кода не стоит, их необходимо обьявлять только там, 

где они начинают(<code> 'работать' </code>) это делается для того, чтобы не забыть что за переменная и для чего она служит.
</div>
<hr/>

Глобальные переменные - это переменные, которые определены вне функции и могут быть использованы в любой части программы. 

Пример:



In [ ]:
# переменная обьявлена вне функции. в основном коде и имеет исходное значение 10
x = 10

# данный кусок кода не делает ничего кроме как распечатывает значение переменной
def my_function():
    print(x)

my_function() # Выведет 10
print(x) # Выведет 10

Важно помнить, что если вы попытаетесь изменить значение глобальной переменной внутри функции, Python создаст локальную переменную с тем же именем, 

и изменения будут применяться только к локальной переменной. 

Если вы хотите изменить глобальную переменную внутри функции, вам нужно использовать ключевое слово `global`. Пример:


In [51]:
# глобальная переменная хранит значение 10
x = 10

# удалите строку с global и запустите ячейку. Потом верните обратно и посмотрите разницу при выводе
def my_function(): 
    global x
    # изменение глобальной переменной
    x = 20

my_function()
print(x) # Выведет 20

20


Константы - это переменные, которые не могут быть изменены в программе. В Python нет специального типа данных для констант, но обычно их имена записываются в верхнем регистре, чтобы отличать их от обычных переменных. 

Пример:

In [ ]:
# Константа
PI = 3.14

# Удаление переменных
Для удаления переменной в Python нужно использовать оператор `del`.

В этом примере переменная `x` сначала присваивается значение 5, затем выводится на экран, 

затем удаляется оператором `del`, и при попытке вывести ее значение возникает ошибка, так как переменная больше не существует

In [52]:
x = 5
print(x) # Выводит 5
del x
print(x) # Ошибка: переменная не определена


5


NameError: name 'x' is not defined

<hr>

<p>Поздравляю, вы прошли свой первый урок и практическую лабораторную работу в Python.
<hr>
